src/features/store/store.controller.ts

```diff
@@ -0,0 +1,138 @@
+import { Request, Response, NextFunction } from 'express';
+import { StoreService } from './store.service';
+import { AppError } from '../../shared/middleware/error-handler';
+
+export class StoreController {
+  constructor(private readonly storeService = new StoreService()) {}
+
+  create = async (req: Request, res: Response, next: NextFunction) => {
+    try {
+      const userId = req.user?.id;
+      const userType = req.user?.type;
+      const { name, address, detailAddress, phoneNumber, content, image } = req.body;
+
+      if (!userId) {
+        throw new AppError(401, '인증이 필요합니다.', 'Unauthorized');
+      }
+
+      if (!userType) {
+        throw new AppError(403, '요청에 필요한 권한 정보가 누락되었습니다.', 'Forbidden');
+      }
+
+      const store = await this.storeService.create(userId, userType, {
+        name,
+        address,
+        detailAddress,
+        phoneNumber,
+        content,
+        image,
+      });
+
+      return res.status(201).json(store);
+    } catch (err) {
+      next(err);
+    }
+  };
+  update = async (req: Request, res: Response, next: NextFunction) => {
+    try {
+      const userId = req.user?.id;
+      const storeId = req.params.storeId;
+      const { name, address, detailAddress, phoneNumber, content, image } = req.body;
+
+      if (!userId) {
+        throw new AppError(401, '인증이 필요합니다.', 'Unauthorized');
+      }
+
+      if (!storeId) {
+        throw new AppError(400, 'storeId가 URL 경로에 필요합니다.');
+      }
+
+      const store = await this.storeService.update(userId, storeId, {
+        name,
+        address,
+        detailAddress,
+        phoneNumber,
+        content,
+        image,
+      });
+
+      return res.status(200).json(store);
+    } catch (err) {
+      next(err);
+    }
+  };
+
+  getStoreDetail = async (req: Request, res: Response, next: NextFunction) => {
+    try {
+      const storeId = req.params.storeId;
+
+      if (!storeId) {
+        throw new AppError(400, 'storeId가 URL 경로에 필요합니다.');
+      }
```

Comment:
아마 이 파일 안에 있는 if (!storeId) 자체가 모두 null 체크용일 것 같은데, 이건
```suggestion
      const storeId = req.params.storeId as string;
```
이런 식으로 단언해도 될 것 같습니다. 애초에 라우터에 id가 없이 들어오면 컨트롤러로 넘어오기 전에 404가 날 거라서요.

Reply:
다만 더 안전하게 하려면 맨 위에
```
type StoreIdParams = { storeId: string };
```
이런 거 넣고
```
getStoreDetail = async (
  req: Request<StoreIdParams>,
  res: Response,
  next: NextFunction,
) => {
  const { storeId } = req.params; // 자동으로 string
  // ...
};
```

이런 식으로 하면 자동으로 string으로 추론할 것 같네요. as를 사용한 단언은 라우트를 수정하다가 에러가 나면 ts가 단언 때문에 잡질 못 하니까요.